cartography/intel/azure/util/credentials.py

```diff
@@ -0,0 +1,173 @@
+
+            arm_credentials, subscription_id, tenant_id = get_azure_cli_credentials(with_tenant=True)
+            aad_graph_credentials, placeholder_1, placeholder_2 = get_azure_cli_credentials(
+                with_tenant=True, resource='https://graph.windows.net',
+            )
+
+            profile = get_cli_profile()
+
+            return Credentials(
+                arm_credentials, aad_graph_credentials, tenant_id=tenant_id,
+                current_user=profile.get_current_account_user(), subscription_id=subscription_id,
+            )
+
+        except HttpResponseError as e:
+            if ', AdalError: Unsupported wstrust endpoint version. ' \
+                    'Current support version is wstrust2005 or wstrust13.' in e.args:
+                logger.error(
+                    f'You are likely authenticating with a Microsoft Account. \
+                    This authentication mode only support Azure Active Directory principal authentication.\
+                    {e}',
+                )
+
+            raise e
+
+    def authenticate_sp(self, tenant_id: str = None, client_id: str = None, client_secret: str = None) -> Any:
+        """
+        Implements authentication for the Azure provider
+        """
+        try:
+
+            # Set logging level to error for libraries as otherwise generates a lot of warnings
+            logging.getLogger('adal-python').setLevel(logging.ERROR)
+            logging.getLogger('msrest').setLevel(logging.ERROR)
+            logging.getLogger('msrestazure.azure_active_directory').setLevel(logging.ERROR)
+            logging.getLogger('urllib3').setLevel(logging.ERROR)
+
+            arm_credentials = ServicePrincipalCredentials(
+                client_id=client_id,
+                secret=client_secret,
+                tenant=tenant_id,
+            )
+
+            aad_graph_credentials = ServicePrincipalCredentials(
+                client_id=client_id,
+                secret=client_secret,
+                tenant=tenant_id,
+                resource='https://graph.windows.net',
+            )
+
+            profile = get_cli_profile()
+
+            return Credentials(
+                arm_credentials, aad_graph_credentials, tenant_id=tenant_id,
+                current_user=profile.get_current_account_user(),
+            )
+
+        except HttpResponseError as e:
+            if ', AdalError: Unsupported wstrust endpoint version. ' \
+                    'Current support version is wstrust2005 or wstrust13.' in e.args:
+                logger.error(
+                    f'You are likely authenticating with a Microsoft Account. \
+                    This authentication mode only support Azure Active Directory principal authentication.\
+                    {e}',
+                )
+
+            raise e
+
+
+class Credentials:
+
+    def __init__(
+        self, arm_credentials: Any, aad_graph_credentials: Any, tenant_id: str = None, subscription_id: str = None,
```

Comment:
I won't block on this now, but we should aim to have these credential objects have typehints of the actual object type and not `Any`.

Reply:
makes sense, will do this as a follow-up.